pkg/test/suite/run.go

```diff
@@ -156,6 +157,9 @@ func runTestCaseWithSharedEnvironment(t *testing.T, suite TestingSuite, suiteOpt
 		stream.ResetInMemoryOutputs()
 		stream.ResetProducerDaemons()
 		kvstore.ResetConfigurableKvStores()
+
+		disp := dispatcher.ProvideDispatcher(suite.Env().Context(), suite.Env().Config(), suite.Env().Logger())
```

Comment:
Why do we need the reset here? Shared environment doesn't mean shared application, so every testcase should be executed in it's own application.. Since the dispatcher is scoped to the application now, there shouldn't be a need to reset it, right?

Reply:
Actually if we use that together with the `WithSharedEnvironment` option, the suites env is not recreated which means that the dispatcher is still in the context. So when we run an `ApiServerTestCase` we run the `SetupApiDefinitions` function  on each test case which will run all the `AddListener` functions which will add multiple listeners to the dispatcher.

Reply:
Mh, okay.. so the api definitions get created with the env context and not the app context? Or is it the same for the integrations test.. Sry, haven't looked at the code for some time.. :grimacing:   
Each app should have it's own context.. If that's not the case, we have to fix that issue..

Reply:
I added a new method `ResetContext` on the test env, an made it just switch the context with a fresh one. As far as I can see, all test cases still run with no problem, so I removed the `Reset` method on the dispatcher again. Thanks for pointing it out.

Reply:
After some discussions with @selm0 , I made the testcases just always inject a new context in each test. This also passes all test cases.